src/Dispatcher/MASA.Contrib.Dispatcher.IntegrationEvents.Dapr/Processor/DeleteLocalQueueExpiresProcessor.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace MASA.Contrib.Dispatcher.IntegrationEvents.Dapr.Processor;
+
+public class DeleteLocalQueueExpiresProcessor : IProcessor
+{
+    private readonly IOptions<DispatcherOptions> _options;
+
+    public DeleteLocalQueueExpiresProcessor(IOptions<DispatcherOptions> options)
+    {
+        _options = options;
+    }
+
+    /// <summary>
+    /// Delete expired events
+    /// </summary>
+    /// <param name="stoppingToken"></param>
+    /// <returns></returns>
+    public async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        LocalQueueProcessor.Default.DeleteAsync(_options.Value.LocalRetryTimes);
+        await Task.Delay(_options.Value.CleaningLocalQueueExpireInterval, stoppingToken);
```

Comment:
Why call delay here? This is a very bad design!

Reply:
If you want use `InfiniteLoopProcessor`, you can add `Sleep` method in IProcessor, then in `InfiniteLoopProcessor` call `_processor.Sleep()`

Reply:
good idea, already adjusted